widgets/AttributesTable/_GraphicsMixin.js

```diff
@@ -648,15 +648,45 @@ define([
             this.clearGraphicsLayer(this.bufferGraphics);
         },
 
-        clearGraphicsLayer: function (layer) {
+        clearGraphicsLayer: function (layer, specificFeatures) {
             if (layer) {
-                layer.clear();
+                if (specificFeatures && specificFeatures.selection && specificFeatures.idProperty) {
+                    this.clearSpecificFeatures(layer, specificFeatures);
+                } else {
+                    layer.clear();
+                }
             }
             this.setToolbarButtons();
             topic.publish(this.attributesContainerID + '/tableUpdated', this);
 
         },
 
+        clearSpecificFeatures: function (layer, specificFeatures) {
+            var selectionFilter = [];
+            for (var key in specificFeatures.selection) {
+                if (key) {
+                    selectionFilter.push(key);
+                }
+            }
+
+            // Tried to to it with dojox.json.query, no success
+            var specificGraphics = layer.graphics.filter(function (graphic) {
+                var idProperty = graphic.attributes[specificFeatures.idProperty];
+                if (idProperty) {
+                    return selectionFilter.indexOf(idProperty.toString()) >= 0;
+                }
+                return false;
+            });
+
+            if (specificGraphics.length > 0) {
+                array.forEach(specificGraphics, function (graphic) {
+                    layer.remove(graphic);
+                });
+
+                topic.publish(this.attributesContainerID + '/recordsRemoved', layer.graphics);
```

Comment:
oh right. I was looking at that backwards. To be consistent, how about publishing the topic like this as it is done elsewhere: 
``` javascript
topic.publish(this.attributesContainerID + '/tableUpdated', this);
```

It would then only need to be called once instead of in the two places as I had suggested.

I recognize your code elsewhere would need to use something like `args.featureGraphics.graphics` instead of just `args`